enlighten/file_io/Configuration.py

```diff
@@ -256,12 +256,13 @@ def dump_keys():
                             continue
 
                         # was it a key-value line?
-                        m = re.match(r"^([A-Za-z0-9_]+) *=", line)
+                        m = re.match(r"^([A-Za-z0-9_ ]+) *=", line)
                         if m and section is not None:
                             # it was a key-value line, so update the line with the current value
-                            key = m.group(1)
-                            self.write(outfile, "%s = %s" % (key, self.get(section, key, raw=True)))
-                            seen[section].add(key)
+                            key = m.group(1).strip()
+                            if key not in seen[section]:
```

Comment:
This ensures we don't re-write duplicate keys we allowed to be loaded from existing files (i.e. cleanup non-conformant files on write)

Reply:
Add to source

Reply:
brief. 
```
# ensure no duplicate keys
```